llvm-spirv/lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -288,6 +288,49 @@ void SPIRVRegularizeLLVMBase::expandVIDWithSYCLTypeByValComp(Function *F) {
       nullptr, &Attrs, true);
 }
 
+// intel/llvm customization
+void SPIRVRegularizeLLVMBase::finishSROACooperativeMatrix(Module *M) {
+  for (auto &F : *M) {
+    if (!F.getName().starts_with("_Z19__spirv_AccessChain"))
+      continue;
+    for (auto I = F.user_begin(), E = F.user_end(); I != E;) {
+      if (auto *CI = dyn_cast<CallInst>(*I++)) {
+        Instruction *Ptr =
+            dyn_cast<Instruction>(CI->getArgOperand(0)->stripPointerCasts());
+        if (!Ptr || !isa_and_nonnull<AllocaInst>(Ptr))
+          continue;
+        StructType *WrapperMatrixTy =
+            dyn_cast<StructType>(cast<AllocaInst>(Ptr)->getAllocatedType());
+        if (!WrapperMatrixTy)
+          continue;
+        TargetExtType *MatrixTy =
+            dyn_cast<TargetExtType>(WrapperMatrixTy->getElementType(0));
+        if (!MatrixTy)
+          continue;
+        StringRef Name = MatrixTy->getName();
+        if (Name.consume_front(kSPIRVTypeName::PrefixAndDelim)) {
+          auto OpCode = SPIRVOpaqueTypeOpCodeMap::map(Name.str());
+          if (OpCode != OpTypeCooperativeMatrixKHR)
+            continue;
+        } else
+          continue;
+
+        AllocaInst *Alloca = nullptr;
+        {
+          IRBuilder Builder(CI);
+          IRBuilderBase::InsertPointGuard IG(Builder);
+          Builder.SetInsertPointPastAllocas(CI->getParent()->getParent());
```

Comment:
```suggestion
          Builder.SetInsertPointPastAllocas(CI->getFunction());
```

? If not, then a comment would be helpful here to help understand where exactly we inserting an alloca

Reply:
Thanks! No reasons, just this W/A was written few months ago and I have reviewed it myself before publishing, but not good enough (with a fresh-code such self-reviews usually go better).